### PR TITLE
[Operator tests]: Attempt to dequarantine dynamic CDI API detection test

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
We should ensure virt-api is stable after picking up the CDI change,
prior to sending requests to it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/operator/operator.go [hunk 1]
```diff
@@ -2407,10 +2407,21 @@ spec:
 			}
 		})
 
-		It("[test_id:3153][QUARANTINE] Ensure infra can handle dynamically detecting DataVolume Support", decorators.Quarantine, func() {
+		It("[test_id:3153] Ensure infra can handle dynamically detecting DataVolume Support", decorators.Quarantine, func() {
```
Comment (brianmcarey) [inline tests/operator/operator.go:2410]: @akalenyu - I think you need to drop the quarantine decorator too - the test is still being skipped.

Comment (akalenyu) [reply tests/operator/operator.go]: Damn right thx

File: tests/operator/operator.go [hunk 2]
```diff
@@ -2443,21 +2454,23 @@ spec:
 
 			}, 240*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
 
-			// wait for virt-api and virt-controller to pick up the change that CDI no longer exists.
-			time.Sleep(30 * time.Second)
+			// wait for virt-api and virt-controller to pick up the change that CDI no longer exists & stabilize
+			time.Sleep(10 * time.Second)
```
Comment (xpivarc) [inline tests/operator/operator.go:2458]: Do we want to do something like
```
Wait for not ready
Wait for ready
```
? Not realizing on knowing the internal timeout?

File: tests/operator/operator.go [hunk 3]
```diff
@@ -2443,21 +2454,23 @@ spec:
...
+			virtAPIStabilized()
 
 			// Verify posting a VM with DataVolumeTemplate fails when DataVolumes
 			// feature gate is disabled
 			By("Expecting Error to Occur when posting VM with DataVolume")
 			_, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), vm)
-			Expect(err).To(HaveOccurred())
+			Expect(err).To(MatchError(ContainSubstring("DataVolume api is not present in cluster")))
```
Comment (xpivarc) [inline tests/operator/operator.go:2465]: +1

**Comment by xpivarc** [comment]
@akalenyu How is it going?

**Comment by akalenyu** [comment]
> @akalenyu How is it going?

Hey, I have not been able to pick up on this, but I believe @ShellyKa13 will be submitting some changes in this area soon
that would not require this test

**Comment by ShellyKa13** [comment]
opened a PR to remove this test alltogether: https://github.com/kubevirt/kubevirt/pull/11010

**Comment by akalenyu** [comment]
/close
https://github.com/kubevirt/kubevirt/pull/11010

**Pull request closed without merging**